### PR TITLE
Instead of logging the URL struct with debug formatting, log it with no formatting since it implements the Display printer

### LINKED ISSUE #2021: Improve watcher URL logging
This is our URL logging at [watcher/src/watcher.rs:191](https://github.com/mobilecoinfoundation/mobilecoin/blob/024d6746629cd8183f0c83451e5deac46e5509d9/watcher/src/watcher.rs#L191).

> ```2022-05-20 01:01:13.949735334 UTC INFO Archive block retrieved for Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("s3-us-west-1.amazonaws.com")), port: None, path: "/mobilecoin.chain/node2.eran.mobilecoin.com/", query: None, fragment: None } 156, mc.app: mobilecoind, mc.module: mc_watcher::watcher, mc.src: watcher/src/watcher.rs:191```

This should print the URL for humanoids, e.g. `https://s3-us-west-1.amazonaws.com/mobilecoin.chain/node2.eran.mobilecoin.com/`, rather than using the structure debug.

### PR DESCRIPTION
**Pull request by pouneh**
### Motivation
Logging URLs in watcher currently is unnecessarily human-unreadable. `as_str()` makes it human readable to decrease cognitive load of reading logs.

Resolves #2021

### CODE REVIEW

File: watcher/src/watcher.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ impl Watcher {
                         log::info!(
                             self.logger,
                             "Archive block retrieved for {:?} {:?}",
```
Comment (eranrund) [inline watcher/src/watcher.rs:193]: You should be able to change this to `"Archive block retrieved for {}: {}",`, I don't think the debug formatters are necessary

**Review by eranrund — APPROVED** [review]

**Review by iamalwaysuncomfortable — APPROVED** [review]

File: watcher/src/watcher.rs (hunk 1 above)
Comment (jcape) [reply watcher/src/watcher.rs:193]: Yep, this will also remove the need for `as_str()`, since `Url` implements `Display` :-)

**Review by jcape — APPROVED** [review]
LGTM, but better to avoid the debug formats in favor of the `Display` printer.

File: watcher/src/watcher.rs (hunk 1 above)
Comment (pouneh) [reply watcher/src/watcher.rs:193]: Sounds good, I'll do that

**Pull request merged**